### PR TITLE
Add password_cipher_mode to encrypt password.

### PR DESCRIPTION
**Pull request by zongzw**
@<reviewer_id>
#### What issues does this address?
Fixes #<issueid>
WIP #<issueid>
...

### CODE REVIEW

File: f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py [hunk 1]
```diff
@@ -389,6 +390,11 @@ def __init__(self, conf, registerOpts=True):
         self.pool_manager = resource_helper.BigIPResourceHelper(
             resource_helper.ResourceType.pool)
 
+        if self.conf.password_cipher_mode:
```
Comment (Niklaus-xie) [inline f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py]: this won't work.   should be encrypt_password?

Comment (zongzw) [reply f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py]: Sorry, not saved before checkin. Change them both to password_cipher_mode.

**Comment by zongzw** [comment]
Manual test: 

```
(Pdb) p self.conf.password_cipher_mode
True
(Pdb) p self.conf.icontrol_password
'YWRtaW4='
(Pdb) l
376  	        self.vs_manager = resource_helper.BigIPResourceHelper(
377  	            resource_helper.ResourceType.virtual)
378  	        self.pool_manager = resource_helper.BigIPResourceHelper(
379  	            resource_helper.ResourceType.pool)
380
381 B->	        if self.conf.password_cipher_mode:
382  	            self.conf.icontrol_password = \
383  	                base64.b64decode(self.conf.icontrol_password)
384  	            self.conf.os_password = base64.b64decode(self.conf.os_password)
385
386  	        try:
(Pdb) n
> /usr/lib/python2.7/site-packages/f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py(383)__init__()
-> base64.b64decode(self.conf.icontrol_password)
(Pdb) n
> /usr/lib/python2.7/site-packages/f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py(384)__init__()
-> self.conf.os_password = base64.b64decode(self.conf.os_password)
(Pdb) p self.conf.icontrol_password
'admin'
(Pdb)
```

**Comment by zongzw** [comment]
Patch readme: 

```
1. Verify the command 'patch' be installed.
    Install it if not:
        a) yum install -y patch, or
        b) wget http://mirror.centos.org/centos/7/os/x86_64/Packages/patch-2.7.1-10.el7_5.x86_64.rpm
            && rpm -i patch-2.7.1-10.el7_5.x86_64.rpm

2. Do patch the file 'cipher-password.patch':
    cd /usr/lib/python2.7/site-packages && patch -p1 cipher-password.patch

3. Update .ini file /etc/neutron/services/f5/f5-openstack-agent.ini:
    1) Add 'password_cipher_mode = True' any where under '[DEFAULT]' session.
    2) Update 'icontrol_password' and 'os_password' value:
        a) Use base64 encode the password:
            echo -n <password string> | base64
        b) Put the encoded strings back-to-place.

4. Restart f5-openstack-agent and check service status:
    systemctl restart f5-openstack-agent
    systemctl status f5-openstack-agent

```

**Comment by zongzw** [comment]
Patch instruction: 

```
1. Verify the command 'patch' be installed.
    Install it if not:
        a) yum install -y patch, or
        b) wget http://mirror.centos.org/centos/7/os/x86_64/Packages/patch-2.7.1-10.el7_5.x86_64.rpm
            && rpm -i patch-2.7.1-10.el7_5.x86_64.rpm

2. Download the  patch file and apply it:
    wget https://patch-diff.githubusercontent.com/raw/F5Networks/f5-openstack-agent/pull/1413.patch -O cipher-password.patch
    cd /usr/lib/python2.7/site-packages && patch -p1 cipher-password.patch

3. Update .ini file /etc/neutron/services/f5/f5-openstack-agent.ini:
    1) Add 'password_cipher_mode = True' any where under '[DEFAULT]' session.
    2) Update 'icontrol_password' and 'os_password' value:
        a) Use base64 encode the password:
            echo -n <password string> | base64
        b) Put the encoded strings back-to-place.

4. Restart f5-openstack-agent and check service status:
    systemctl restart f5-openstack-agent
    systemctl status f5-openstack-agent

```

**Comment by zongzw** [comment]
> seems to me you could abandon this PR, since you submitted another one which also include these same modifications.

This is only for patch? how about it?

**Pull request merged**